Core/src/org/sleuthkit/autopsy/centralrepository/AddEditCentralRepoCommentAction.java

```diff
@@ -0,0 +1,144 @@
+     *
+     * @param correlationAttribute The correlation attribute to modify.
+     * @param displayName          The text for the menu item.
+     */
+    private AddEditCentralRepoCommentAction(CorrelationAttribute correlationAttribute, String displayName) {
+        super(displayName);
+        this.correlationAttribute = correlationAttribute;
+    }
+
+    /**
+     * Private constructor to create an instance given an AbstractFile.
+     *
+     * @param file        The file from which a correlation attribute to modify
+     *                    is derived.
+     * @param displayName The text for the menu item.
+     */
+    private AddEditCentralRepoCommentAction(AbstractFile file, String displayName) throws EamDbException, NoCurrentCaseException, TskCoreException {
+
+        super(displayName);
+        correlationAttribute = EamArtifactUtil.getCorrelationAttributeFromContent(file);
+        if (correlationAttribute == null) {
+            addToDatabase = true;
+            correlationAttribute = EamArtifactUtil.makeCorrelationAttributeFromContent(file);
+        }
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent event) {
+        addEditCentralRepoComment();
+    }
+
+    /**
+     * Create a Add/Edit dialog for the correlation attribute file instance
+     * comment. The comment will be updated in the database if the file instance
+     * exists there, or a new file instance will be added to the database with
+     * the comment attached otherwise.
+     */
+    public void addEditCentralRepoComment() {
+        CentralRepoCommentDialog centralRepoCommentDialog = new CentralRepoCommentDialog(correlationAttribute);
+        centralRepoCommentDialog.display();
+
+        if (centralRepoCommentDialog.isCommentUpdated()) {
+            EamDb dbManager;
+
+            try {
+                dbManager = EamDb.getInstance();
+
+                if (addToDatabase) {
+                    dbManager.addArtifact(correlationAttribute);
+                } else {
+                    dbManager.updateAttributeInstanceComment(correlationAttribute);
+                }
+            } catch (EamDbException ex) {
+                logger.log(Level.SEVERE, "Error connecting to Central Repository database.", ex);
+            }
+        }
+    }
+
+    /**
+     * Create an instance labeled "Add/Edit Central Repository Comment" given an
+     * AbstractFile. This is intended for the result view.
+     *
+     * @param file The file from which a correlation attribute to modify is
+     *             derived.
+     *
+     * @return The instance.
+     * 
+     * @throws EamDbException
+     * @throws NoCurrentCaseException
+     * @throws TskCoreException
+     */
+    @Messages({"AddEditCentralRepoCommentAction.menuItemText.addEditCentralRepoComment=Add/Edit Central Repository Comment"})
+    public static AddEditCentralRepoCommentAction createAddEditCentralRepoCommentAction(AbstractFile file)
```

Comment:
It might be cleaner for this (or a method down the chain) to wrap all the exception types into one instead of throwing three different types that you then have to check for individually.